src/sagemaker/jumpstart/factory/model.py

```diff
@@ -740,27 +760,41 @@ def _add_config_name_to_deploy_kwargs(
         ValueError: If the instance_type is not supported with the current config.
     """
 
-    specs = verify_model_region_and_return_specs(
-        model_id=kwargs.model_id,
-        version=kwargs.model_version,
-        scope=JumpStartScriptScope.INFERENCE,
-        region=kwargs.region,
-        tolerate_vulnerable_model=kwargs.tolerate_vulnerable_model,
-        tolerate_deprecated_model=kwargs.tolerate_deprecated_model,
-        sagemaker_session=kwargs.sagemaker_session,
-        model_type=kwargs.model_type,
-        config_name=kwargs.config_name,
-    )
+    # we need to create a default JS session (without custom user agent)
+    # in order to retrieve config name info
+    temp_session = kwargs.sagemaker_session or DEFAULT_JUMPSTART_SAGEMAKER_SESSION
```

Comment:
Not sure if I follow this, if kwargs.sagemaker_session is None, don't we fallback to the default session [here](https://github.com/aws/sagemaker-python-sdk/pull/4806/files#diff-40606ba8ca26af6a8ca7507075a65dddd70b0da2d3b126561beef72cd9780392R167)?

Reply:
yes, but the problem is we need a session in order to derive what the config name is. 

For the session to get the config name, we use the `DEFAULT_JUMPSTART_SAGEMAKER_SESSION` without any special user agent, unless a custom session is provided